### PR TITLE
HWKMETRICS-113 - refactor core model classes

### PR DESCRIPTION
**Pull request by jsanda**
This is not quite ready to be merged, but I would like to start discussing the changes. There was a lot of implementation details bleeding into the API. This PR addresses those issues. This PR also starts to add back some model classes into the hawkular-metrics-api-jaxrs module. The other option would be for the REST API to have a direct dependency on core-impl which I think is a bad idea.

### CODE REVIEW

**Comment by tsegismont** [comment]
> This PR also starts to add back some model classes into the hawkular-metrics-api-jaxrs module
+1

File: api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/influx/InfluxSeriesHandler.java [hunk 1]
```diff
@@ -119,7 +116,7 @@ public void write(
             @Suspended AsyncResponse asyncResponse, @PathParam("tenantId") String tenantId,
             List<InfluxObject> influxObjects
     ) {
-
+        LOG.info("WRITE REQUEST");
```
Comment (tsegismont) [inline api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/influx/InfluxSeriesHandler.java]: Something you forgot to remove after debugging?

File: core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/Functions.java [hunk 2]
```diff
@@ -16,24 +16,25 @@
...
+import com.datastax.driver.core.utils.UUIDs;
 import com.google.common.base.Function;
+import org.hawkular.metrics.core.api.AggregationTemplate;
+import org.hawkular.metrics.core.api.AvailabilityDataPoint;
+import org.hawkular.metrics.core.api.AvailabilityType;
+import org.hawkular.metrics.core.api.GaugeDataPoint;
+import org.hawkular.metrics.core.api.Interval;
+import org.hawkular.metrics.core.api.MetricType;
+import org.hawkular.metrics.core.api.Tenant;
+import org.joda.time.Duration;
```
Comment (tsegismont) [inline core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/Functions.java]: John there's still a lot of those import changes

File: rest-tests/src/test/groovy/org/hawkular/metrics/rest/CassandraBackendITest.groovy [hunk 3]
```diff
@@ -350,7 +350,7 @@ class CassandraBackendITest extends RESTTest {
     )
   }
 
-  @Test
+//  @Test
```
Comment (tsegismont) [inline rest-tests/src/test/groovy/org/hawkular/metrics/rest/CassandraBackendITest.groovy]: Something you forgot to uncomment? Or still under works?

File: rest-tests/src/test/groovy/org/hawkular/metrics/rest/CassandraBackendITest.groovy [hunk 4]
```diff
@@ -434,7 +434,7 @@ class CassandraBackendITest extends RESTTest {
...
```
Comment (tsegismont) [inline rest-tests/src/test/groovy/org/hawkular/metrics/rest/CassandraBackendITest.groovy]: Something you forgot to uncomment? Or still under works?

File: core/metrics-core-impl/src/test/java/org/hawkular/metrics/core/impl/MetricsServiceITest.java [hunk 5]
```diff
@@ -927,14 +927,14 @@ public void getPeriodsAboveThreshold() throws Exception {
         return ImmutableList.copyOf(observable.toBlocking().toIterable());
     }
 
-    private void assertMetricEquals(Metric actual, Metric expected) {
-        assertEquals(actual, expected, "The metric doe not match the expected value");
-        assertEquals(actual.getData(), expected.getData(), "The data does not match the expected values");
-    }
+//    private void assertMetricEquals(Metric actual, Metric expected) {
```
Comment (tsegismont) [inline core/metrics-core-impl/src/test/java/org/hawkular/metrics/core/impl/MetricsServiceITest.java]: Remove if it's no longer used

**Comment by FilipB** [comment]
retest this please

**Comment by FilipB** [comment]
retest this please

**Comment by FilipB** [comment]
retest this please

**Comment by FilipB** [comment]
retest this please

**Comment by tsegismont** [comment]
@jsanda Travis build fails in jaxrs module on checkstyle execution because of a line too long

**Comment by jsanda** [comment]
I don't want to you have to bed :) I can try to update my IDE settings tomorrow. I am not going to do it right now though because last time I wound up burning a good hour on it.

**Comment by tsegismont** [comment]
I'm done. This is a good one. Ping me when needed for more reviewing / merging.

**Comment by tsegismont** [comment]
Are you done with the PR?

**Comment by jsanda** [comment]
I think it is ready for final review/merge. I made one more big change. I made ```DataPoint``` a concrete class. I also created [HWKMETRICS-123](https://issues.jboss.org/browse/HWKMETRICS-123) for the swagger doc issue.

**Comment by tsegismont** [comment]
Changes looks good to me. I need to merge manually because there's a small conflict in the Influx handler due to PR #246 , merged yesterday.

**Comment by tsegismont** [comment]
Manually merged. Thanks John.

**Pull request merged**